qiskit_nature/operators/second_quantization/quadratic_hamiltonian.py

```diff
@@ -0,0 +1,351 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021, 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""The QuadraticHamiltonian class."""
+
+from __future__ import annotations
+
+from typing import Optional
+
+import numpy as np
+import scipy.linalg
+from qiskit.quantum_info.operators.mixins import TolerancesMixin
+from qiskit_nature.operators.second_quantization.fermionic_op import FermionicOp
+
+
+def _is_hermitian(mat: np.ndarray, rtol: float = 1e-5, atol: float = 1e-8) -> bool:
+    return np.allclose(mat, mat.T.conj(), rtol=rtol, atol=atol)
+
+
+def _is_antisymmetric(mat: np.ndarray, rtol: float = 1e-5, atol: float = 1e-8) -> bool:
+    return np.allclose(mat, -mat.T, rtol=rtol, atol=atol)
+
+
+class QuadraticHamiltonian(TolerancesMixin):
+    r"""A Hamiltonian that is quadratic in the fermionic ladder operators.
+
+    A quadratic Hamiltonian is an operator of the form
+
+    .. math::
+        \sum_{p, q} M_{pq} a^\dagger_p a_q
+        + \frac12 \sum_{p, q}
+            (\Delta_{pq} a^\dagger_p a^\dagger_q + \text{h.c.})
+        + \text{constant}
+
+    where :math:`M` is a Hermitian matrix and :math:`\Delta` is an antisymmetric matrix.
+    """
+
+    def __init__(
+        self,
+        hermitian_part: Optional[np.ndarray] = None,
+        antisymmetric_part: Optional[np.ndarray] = None,
+        constant: float = 0.0,
+        num_modes: Optional[int] = None,
+        validate: bool = True,
+        rtol: float = 1e-5,
+        atol: float = 1e-8,
+    ) -> None:
+        r"""Initialize a QuadraticHamiltonian.
```

Comment:
This text can/should be removed `Initialize a QuadraticHamiltonian.` We had other places where things like X constructor is put as text. Since this is Python this is what the init method its not necessary to state the obvious. As the class docstring is merged with the docstring here when its comes out in the docs we can go straight to args I think

![image](https://user-images.githubusercontent.com/40241007/161825178-35b9f2da-51bb-426d-8c01-18af45d10649.png)

Reply:
Done.